src/app.md

```diff
@@ -725,6 +726,69 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-semver-latest]: #ex-semver-latest
+<a name="ex-semver-latest"></a>
+## Find the latest version satisfying given range
+[![semver-badge]][semver] [![cat-config-badge]][cat-config]
+
+Given a list of version &strs, finds the latest [`semver::Version`] that satisfying a given [`semver::VersionReq`] using [`VersionReq::matches`].
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate semver;
+
+use semver::{Version, VersionReq};
+#
+# error_chain! {
+#     foreign_links {
+#         SemVer(semver::SemVerError);
+#         SemVerReq(semver::ReqParseError);
+#     }
+# }
+
+fn find_max_matching_version<'a, I>(version_req_str: &str, iterable: I) -> Result<Option<Version>>
```

Comment:
The semantic of the return type `Result<Option<T>>` looks off to me. However, I have no better suggestion. Does it mean something specific?

Reply:
> Does it mean something specific?

Yep. Typically `Result` is used to handle possibility of error while `Option` expresses possibility of nonexistence. So as long as nonexistence of searched value is not considered a hard error it would be best expressed as an `Option` while having to deal with parsing errors leaves us with `Result` so `Result<Option<T>>` gives the reader of our API a clear indication that the operation can both fail as well as yield no usable return value even if there is no error.

Reply:
Thanks @budziq for your explanation. I appreciate it!